arviz/stats/diagnostics.py

```diff
@@ -510,13 +511,22 @@ def geweke(ary, first=0.1, last=0.5, intervals=20):
         last_slice = ary[int(end - last * (end - start)) :]
 
         z_score = first_slice.mean() - last_slice.mean()
-        z_score /= np.sqrt(first_slice.var() + last_slice.var())
+        if numba_check():
```

Comment:
This comment is not required. While this if else pattern works I challenge us both to come up with a "nicer" pattern. I have some ideas and we can talk together about this.

Again don't worry about it right now but did want to make the comment

Reply:
@Ban-zee In the last 20 minutes I thought about this and think we should change on this PR before we move.

We shouldn't do a numba check everytime a function is called. I think a preferred pattern would be to check if Numba is installed once when ArviZ is loaded and then use that constant throughout the entire program.

I also think this will fit in well because users can optionally turn numba off, even if they have it installed.

If you want I can set some time to talk about this synchronously if you want :)

Reply:
This can be done. I have a few questions in mind and I am up for discussion anytime. :)